app/src/main/java/ar/com/wolox/android/training/ui/training/LoginActivity.java

```diff
@@ -0,0 +1,137 @@
+package ar.com.wolox.android.training.ui.training;
+
+import android.content.ActivityNotFoundException;
+import android.content.ComponentName;
+import android.content.Intent;
+import android.net.Uri;
+import android.os.Bundle;
+import android.text.TextUtils;
+import android.util.Log;
+import android.util.Patterns;
+import android.view.View;
+import android.widget.Button;
+import android.widget.TextView;
+
+import androidx.appcompat.app.AppCompatActivity;
+
+import com.google.android.material.textfield.TextInputEditText;
+import com.google.android.material.textfield.TextInputLayout;
+
+import ar.com.wolox.android.R;
+
+/**
+ *   Login activity
+ */
+public class LoginActivity extends AppCompatActivity implements View.OnClickListener {
+
+    private static final String TAG = "LOGIN_ACTV";
+    public static final String URL_TYC = "https://www.wolox.com.ar/";
+
+    private TextInputEditText mEmailTxt;
+    private TextInputEditText mPassTxt;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_login);
+
+        Button mBtnLogin = findViewById(R.id.btn_login);
+        mBtnLogin.setOnClickListener(this);
+        Button mBtnSingup = findViewById(R.id.btn_singup);
+        mBtnSingup.setOnClickListener(this);
+        TextView mTvTyc = findViewById(R.id.tyc_txt);
+        mTvTyc.setOnClickListener(this);
+
+        mEmailTxt = findViewById(R.id.user_text);
+        mPassTxt = findViewById(R.id.pass_text);
+    }
+
+    @Override
+    public void onClick(View view) {
+        switch (view.getId()) {
+            case R.id.btn_login:
+                toMain();
```

Comment:
This would happened on futures PRs, but since you already did it I'll mark it 😅. On the MVP architecture we have a view, a presenter and the model. Here at least, we need the view abstraction and the presenter. The most important thing about MVP is that the view informs events to the presenter and the presenter is who knows all the business logic and will order what to do to the view. In the project there is an `Example` where you can see how to implement this.

All this file should be refactored to fulfil this architecture, so I won't comment until you change it 😉